client-spark/common/src/main/java/org/apache/spark/shuffle/celeborn/SortBasedPusher.java

```diff
@@ -169,6 +238,7 @@ public long pushData() throws IOException {
                   numPartitions);
           mapStatusLengths[currentPartition].add(bytesWritten);
           afterPush.accept(bytesWritten);
+          memoryThresholdManager.updateStats(offSet, offSet == pushBufferMaxSize);
```

Comment:
I think for most cases `offSet == pushBufferMaxSize` is false, but can be close. So maybe just comparing with `pushBufferMaxSize / (1 + factor)` is fine.

Reply:
actually I tried to compare `pushBufferMaxSize / (1 + factor)` and I found my unit test always failed in that case , the reason is that we actually triggered  line 251 `memoryThresholdManager.updateStats(offSet, true);` for many times with a partially filled buffer....

I agree that `offSet == pushBufferMaxSize` is rarely true but to make the precise decision about whether to increase the buffer, we still need tracking "shouldPushedBytes" and "shouldPushedCount", therefore we still need a switch in the method `updateStats` whether we want to update these two values

Reply:
Makes sense, thanks!